packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ClientMethodTemplate.java

```diff
@@ -162,13 +173,25 @@ protected static void addOptionalAndConstantVariables(JavaBlock function, Client
                 parameterWireType = ClassType.STRING;
             }
 
-            // If the parameter isn't required and the client method only uses required parameters, optional
-            // parameters are omitted and will need to instantiated in the method.
-            boolean optionalOmitted = clientMethod.getOnlyRequiredParameters() && !parameter.isRequired();
+            // If the parameter isn't required and the client method only uses required parameters,
```

Comment:
`the client method only uses required parameters`: just trying to understasnd the meaning. Does it mean in the client method with only required parameters, we need to initialize the optional parameters value? But I thought in client method with full parameters, we need to initialize optional parameters if user does not provide as well. 
Would be awesome if we  could add an example (better in javadoc) to explain the case we want to handle in the logics from line 176-190?